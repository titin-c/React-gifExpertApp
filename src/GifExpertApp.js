
import React,  {useState}  from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp  = ({ defaultCategories = ['Popeye'] } ) =>{

    // const defaultCategories = ['Goku', 'pikachu', 'Vegetta'];
    
    //const [categories, setCategories] = useState('pikachu');
    const [categories, setCategories] = useState(defaultCategories);

    // función handleAdd
    // const handleAdd = () =>{

    //dos opciones que cogen todas las categorías y le añade heidy al final

    // //setCategories( [...categories, 'Heidy']); 
    // setCategories( cats => [...cats, 'Heidy']); 

    //}

    return (
        <>
        <h1>GifExpertApp</h1>
        {/* añadimos el componente AddCategory*/}
        <AddCategory setCategories = {setCategories}/>
                
            {
                //mapeamos el array categories para que nos devuelva su contenido
                categories.map( category =>(
                    /* añadimos el componente GifGrid*/
                    <GifGrid 
                    // el key es obligatorio y único
                    key={category}
                    category={category} 
                    />
                ))
            }
        
        </>
    );
}

export default GifExpertApp;
