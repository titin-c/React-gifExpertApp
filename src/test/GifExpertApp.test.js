
import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from "../GifExpertApp";


describe('Pruebas sobre el componente <GifExpertApp />', () => {
  
    
    test('debe de mostrarse correctamente', () => {

        const wrapper = shallow( <GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
       
    });

    test('debe mostrar una lista de categorÃ­as', () => {
      
      const categories = ['Pikatchu', 'Goku'];
      const wrapper = shallow( <GifExpertApp defaultCategories={ categories}/>);

      expect(wrapper).toMatchSnapshot();
      expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });
    
});