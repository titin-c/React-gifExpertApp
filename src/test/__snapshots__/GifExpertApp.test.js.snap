// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifExpertApp /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
</Fragment>
`;

exports[`Pruebas sobre el componente <GifExpertApp /> debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="Pikatchu"
    key="Pikatchu"
  />
  <GifGrid
    category="Goku"
    key="Goku"
  />
</Fragment>
`;
